sockeye/utils.py

```diff
@@ -32,11 +32,22 @@
 
 logger = logging.getLogger(__name__)
 
+def check_condition(condition: bool, error_message: str, exception: Exception = RuntimeError):
+    """
+    Check the condition and if it is not met, exit with the given error message
+    and error_code, similar to assertions.
+
+    :param condition: Condition to check.
+    :param error_message: Error message to show to the user.
+    :param error_code: Error code to return to the system.
+    """
+    if not condition:
+        raise exception(error_message)
```

Comment:
it'd just define the scope of the exception more clearly, but no strong feelings about it